### PR TITLE
Tags and Extra exposed as IDictionary (instead of IReadOnlyDictionary)

### PR DESCRIPTION
**Pull request by jamescrosswell**
Resolves https://github.com/getsentry/sentry-dotnet/issues/2273

### CODE REVIEW

File: src/Sentry/Internal/ScopeSyncingDictionary.cs [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+{
+    private readonly IDictionary<K, V> _innerDictionary;
+    protected readonly SentryOptions Options;
+
+    public ScopeSyncingDictionary(
+        IDictionary<K, V> innerDictionary,
+        SentryOptions options
+        )
+    {
+        _innerDictionary = innerDictionary ?? throw new ArgumentNullException(nameof(innerDictionary));
```
Comment (bitsandfoxes) [inline src/Sentry/Internal/ScopeSyncingDictionary.cs:13]: This is never getting caught, right? Should we log this instead?

**Review by bruno-garcia — CHANGES_REQUESTED** [review]
Lets please not to this. `SetTag` is the API in all Sentry SDKs, and it's what we exposed for 5 and a half years here. We gain nothing in changing this. In fact I'd argue we lose in exposing the IDictionary API which exposes things in a way we might not have expected/planned

**Comment by bruno-garcia** [comment]
We had a bit of a race condition here but I close this issue 2 days ago too: https://github.com/getsentry/sentry-dotnet/issues/2273

**Comment by jamescrosswell** [comment]
> Lets please not to this. `SetTag` is the API in all Sentry SDKs, and it's what we exposed for 5 and a half years here. We gain nothing in changing this. In fact I'd argue we lose in exposing the IDictionary API which exposes things in a way we might not have expected/planned

I'm OK with that. It would be good to make these calls before writing the solution/implementation though. I was basically just going through the remaining issues on the 4.0.0 milestone... we could maybe do a bit of "backlog grooming" every now and then to avoid writing code that will never get used (like this).

**Pull request closed without merging**